terraform-aws-modules/terraform-aws-vpc
# fix: use filter for getting service type for S3 endpoint and update to allow s3 to use interface endpoint types

## Description
- use filter for getting service type for S3 endpoint and update to allow s3 to use interface endpoint types
- reduce min supported version of AWS provider back to 2.70.0 with these changes

## Motivation and Context
- there was a change made to avoid conflicts between the newly introduced S3 `Interface` endpoint and instead keep the module at the default setting of `Gateway` for backwards comparability. However, this caused two issues:
1. Users need to upgrade their version of the AWS provider to have the attribute type `s3_endpoint_type` available to them
2. If users were to switch to an `Interface` type for S3, it would fail

This PR attempts to solve both of those problems and the `examples/complete-vpc` has been updated to validate and demonstrate this change.

Ref: #582 
Closes #575
Closes #578
Closes #585 
Closes #587 

## Breaking Changes
- No

## How Has This Been Tested?
1. Checkout `master` locally at time of writing
2. `cd examples/complete-vpc && terraform init && terraform apply`
3. Checkout this PR locally
4. `terraform apply` -> S3 endpoint type successfully changes to type `Interface` following the norms established
5. I also re-applied `master` to set back to "current" and again checked out my PR but setting the example back to what it is currently on `master` and no changes which means that backwards compatibility should be preserved

I also tested by setting the `examples/simple-vpc` AWS version to `v2.70.0` and validating that it works at that version as well still
